src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,361 @@
+      suppressAutoSize: false,
+      suppressSizeToFit: false,
+    },
+    {
+      field: 'source_type_id',
+      filter: 'string',
+      headerName: 'Source type',
+      resizable: true,
+      sortable: true,
+      suppressAutoSize: false,
+      suppressSizeToFit: false,
+      valueFormatter: params => {
+        const sourceTypeName = getEventSourceTypeName(params?.value, $externalSourceTypes);
+        return sourceTypeName ? sourceTypeName : '';
+      },
+    },
+    {
+      field: 'derivedEventTotal',
+      filter: 'number',
+      headerName: 'Derived Events in Derivation Group',
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      valueFormatter: params => {
+        return params?.value.length;
+      },
+    },
+    {
+      cellRenderer: (params: DerivationGroupCellRendererParams) => {
+        var input = document.createElement('input');
+        input.type = 'checkbox';
+        input.checked = $selectedPlanDerivationGroupIds.includes(params?.data?.id ?? -1);
+        input.addEventListener('click', event => {
+          if (event?.target && params.data) {
+            if ((event.target as any).checked) {
+              // insert
+              effects.insertDerivationGroupForPlan(params.data.id, $plan, user);
+              if ($derivationGroupPlanLinkError !== null) {
+                console.log($derivationGroupPlanLinkError);
+                console.log('Failed to link derivation group & plan.');
+              } else {
+                // Insert all the external event types from the derivation group to the timeline filter
+                const derivationGroup = $derivationGroups.find(
+                  derivationGroup => derivationGroup.id === params?.data?.id,
+                );
+                if (derivationGroup !== undefined) {
+                  externalEventLayers?.forEach(externalEventLayer => {
+                    if (externalEventLayer.filter.externalEvent !== undefined) {
+                      externalEventLayer.filter.externalEvent.event_types =
+                        externalEventLayer.filter.externalEvent.event_types.concat(derivationGroup.event_types);
+                      externalEventLayer.filter.externalEvent.event_types =
+                        externalEventLayer.filter.externalEvent.event_types.filter(
+                          (val, ind, arr) => arr.indexOf(val) === ind,
+                        ); // uniqueness
+                    }
+                  });
+                }
+              }
+            } else {
+              // delete
+              effects.deleteDerivationGroupForPlan(params.data.id, $plan, user);
+              if ($derivationGroupPlanLinkError !== null) {
+                console.log($derivationGroupPlanLinkError);
+                console.log('Failed to unlink derivation group & plan.');
+              }
+            }
+          }
+        });
+        return input;
+      },
+      enableCellChangeFlash: true,
```

Comment:
The justification for this was so that if multiple users were using the same AERIE instance, then if there was a change to what is associated with a plan by another user, this would flash to indicate the change. That being said, we ultimately decided against this as it proved more confusing than not. This flash behavior no longer exists!